core/src/test/scala/unit/kafka/server/ReplicaManagerTest.scala

```diff
@@ -2810,4 +2811,132 @@ class ReplicaManagerTest {
           Replicas.NONE, Replicas.NONE, 2, 123, 456)))),
     replicaManager.calculateDeltaChanges(TEST_DELTA))
   }
+
+  @Test
+  def testPartitionFetchStateUpdatesWithTopicIdAdded(): Unit = {
+    val aliveBrokersIds = Seq(0, 1)
+    val replicaManager = setupReplicaManagerWithMockedPurgatories(new MockTimer(time),
+      brokerId = 0, aliveBrokersIds)
+    try {
+      val tp = new TopicPartition(topic, 0)
+      val replicas = aliveBrokersIds.toList.map(Int.box).asJava
+
+      def leaderAndIsrRequest(epoch: Int, topicIds: util.Map[String, Uuid]): LeaderAndIsrRequest =
```

Comment:
You use this method in the two tests. Should we pull it out into an helper method? Also, it might be worth checking if we already have such helper in the class.

Reply:
Seems like we have a few in each of the tests with similar or slightly different parameters. Maybe it is ok to rewrite parameters to not have to rewrite the construction? Will take a closer look.

Reply:
Ah I see there is an already existing one. I modified my two tests to use it. If you could take a second look and make sure the parameters match, that would be super helpful. :)